### PR TITLE
Use starknet-types-core Felt

### PR DESCRIPTION
**Pull request by pefontana**
# Use starknet-types-core Felt

* Replace cairo_felt::Felt252 with starknet_types_core::felt::Felt. We still need to keep the cairo_felt::Felt252 dependencies for testing vs the cairo-vm (we use cairo-lang-runner to run cairo programs with the vm, that uses an old vm version with the old Felt)
* Rename Felt252 with Felt

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

**Review by Oppen — APPROVED** [review]

File: tests/arrays.rs [hunk 1]
```diff
@@ -82,8 +85,8 @@ proptest! {
     fn array_get_test_proptest(value in any_felt252(), idx in 0u32..26) {
         let program = &ARRAY_GET;
         let result_vm = run_vm_program(program, "run_test", &[
-            Arg::Value(value.clone()),
-            Arg::Value(Felt252::new(idx))
+            Arg::Value(DeprecatedFelt::from_bytes_be(&value.clone().to_bytes_be())),
```
Comment (azteca1998) [inline tests/arrays.rs:88]: I'm not sure we need a clone here.

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
+        if byte == 0 {
+            is_end = true;
+        } else if is_end || !byte.is_ascii() {
+            return None;
+        } else {
+            as_string.push(byte as char);
+        }
+    }
+    Some(as_string)
+}
```
[the remark this answers is not in the document]
Comment (pefontana) [reply runtime/src/lib.rs:21]: I think that `String::from_utf8(value.to_vec()).ok()` can return different things.
I don't know what you mean with` value.to_vec()`

File: tests/arrays.rs [hunk 3]
```diff
@@ -82,8 +85,8 @@ proptest! {
     fn array_get_test_proptest(value in any_felt252(), idx in 0u32..26) {
```
[the remark this answers is not in the document]
Comment (pefontana) [reply tests/arrays.rs:85]: Thanks @azteca1998 !
Good catch

File: tests/arrays.rs (hunk 1 above)
Comment (pefontana) [reply tests/arrays.rs:88]: done!

File: tests/common.rs [hunk 4]
```diff
@@ -669,8 +669,8 @@ pub fn compare_outputs(
 pub const FIELD_HIGH: u128 = (1 << 123) + (17 << 64); // this is equal to 10633823966279327296825105735305134080
 pub const FIELD_LOW: u128 = 1;
 
-/// Returns a [`Strategy`] that generates any valid Felt252
-pub fn any_felt252() -> impl Strategy<Value = Felt252> {
+/// Returns a [`Strategy`] that generates any valid Felt
+pub fn any_felt252() -> impl Strategy<Value = Felt> {
```
[the remark this answers is not in the document]
Comment (pefontana) [reply tests/common.rs:658]: done!

File: tests/common.rs [hunk 5]
```diff
@@ -696,11 +696,11 @@ pub fn any_felt252() -> impl Strategy<Value = Felt252> {
...
         .prop_map(|value: FieldElement<MontgomeryBackendPrimeField<_, 4>>| {
-            Felt252::from_bytes_be(&value.to_bytes_be())
+            Felt::from_bytes_be(&value.to_bytes_be())
         })
 }
 
-/// Returns a [`Strategy`] that generates any nonzero Felt252
-pub fn nonzero_felt252() -> impl Strategy<Value = Felt252> {
+/// Returns a [`Strategy`] that generates any nonzero Felt
+pub fn nonzero_felt252() -> impl Strategy<Value = Felt> {
```
[the remark this answers is not in the document]
Comment (pefontana) [reply tests/common.rs:689]: Done!

File: src/utils.rs [hunk 6]
```diff
@@ -887,7 +887,7 @@ pub mod test {
         assert_eq!(get_integer_layout(512).align(), 8);
     }
 
-    /// Ensures that the host's `felt252` is compatible with its compiled counterpart.
+    /// Ensures that the host's `Felt` is compatible with its compiled counterpart.
     #[test]
     #[ignore]
     fn test_alignment_compatibility_felt252() {
```
[the remark this answers is not in the document]
Comment (pefontana) [reply src/utils.rs:893]: done!

File: runtime/src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (azteca1998) [reply runtime/src/lib.rs:21]: Sorry, I meant `value.to_bytes_be().to_vec()`.

File: README.md
[the remark this answers is not in the document]
Comment (pefontana) [reply README.md:1]: Thanks @azteca1998 !
I renamed all the corresponding Felts252 with Felt

[the remark this answers is not in the document]
Comment (pefontana) [reply README.md:1]: Just to be clear here.
Yes, we renamed the Felt252 to Felt in the starknet_types_core crate.
I still leave the JITValue::Felt252 name unchanged.
Maybe we can renamed it in another PR

File: runtime/src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (pefontana) [reply runtime/src/lib.rs:21]: Yes, I try it and they return different things

**Review by edg-l — APPROVED** [review]

**Pull request merged**